src/libraries/Microsoft.PowerFx.Core/Parser/TexlParser.cs

```diff
@@ -56,6 +56,120 @@ private TexlParser(IReadOnlyList<Token> tokens, Flags flags)
             _flags = flags;
         }
 
+        public static ParseUDFsResult ParseUDFsScript(string script, CultureInfo loc = null)
+        {
+            Contracts.AssertValue(script);
+            Contracts.AssertValueOrNull(loc);
+
+            var formulaTokens = TokenizeScript(script, loc, Flags.NamedFormulas);
+            var parser = new TexlParser(formulaTokens, Flags.NamedFormulas);
+
+            return parser.ParseUDFs(script);
+        }
+
+        private ParseUDFsResult ParseUDFs(string script)
+        {
+            var udfs = new List<UDF>();
+            ParseTrivia();
+
+            while (_curs.TokCur.Kind != TokKind.Eof)
+            {
+                var args = new HashSet<Arg>();
```

Comment:
Bit problem with using a HashSet here is that we are not checking to see if the arg exists before adding to this collection. We need to give a custom error message in the case where the user defines two arguments with the same name.

Reply:
Changed to using a dictionary and returning a texl error if it encounters duplicates.